### DISCUSSION TITLE
[Bug]: display shutting down when external notification set to pwm, vibrate and buzzer

### DISCUSSION THREAD

**Discussion by adamstern2000** [meshtastic/firmware#5177] [Troubleshooting]
### Category

Other

### Hardware

T-Beam S3

### Firmware Version

2.5.6

### Description

Affects 2.4.3, 2.5.4, 2.5.6 and probably many more.
I am using a t-beam s3 core with 5v buzzer and 3.3v vibrator.

external notification settings:
Enabled true
Alert bell true
Alert message true
Use pwm true
Active true
Output pin 3
Output duration unset
Nag unset
Alert buzzer on bell true
Alert vibra on bell true
Alert buzzer on message true
Alert vibra on message true
Output pin buzzer pin 3
Output pin vibra 46

### Relevant log output

```Shell
-
```

**Comment by garthvh** [comment]
Pretty likely you have a GPIO pin conflict with the buzzer.

**Reply by adamstern2000** [reply]
I don’t think this should have been moved from issues at this point.  There is no reasonable explanation why it would say shutting down on the screen and not shut down.  Obviously something strange is happening.

**Comment by todd-herbert** [comment]
Are you certain that your device is the T-Beam S3 Core? If your device is instead a ~~T-Beam S3 Supreme~~ T3S3, the pin conflict would be with IO3, which is used by the LoRa radio.

**Comment by adamstern2000** [comment]
Thanks Todd.  That is interesting- I’ll have to fix that.  But I’ve also tested on ip38, and it does the exact same thing.

and you are right- it’s the supreme.  I loaded the firmware for s3 core.  Maybe that’s the issue.  Which firmware for t-beam supreme?

**Comment by adamstern2000** [comment]
When I disable the pwm, the issue goes away.  But the beep becomes a chirp and the vibration doesn’t work.  

The vibration has always been on the same.pin.  The buzzer has been on 38 and 3 - depending on my build, and it happens on both configs.

**Comment by todd-herbert** [comment]
Sorry, I'm confusing myself slightly. I meant to say ["T3S3"](https://www.lilygo.cc/products/t3s3-v1-0) earlier, which is different from ["T-Beam Supreme"](https://www.lilygo.cc/products/t-beam-supreme-meshtastic). As best I can tell there is no pin conflict with the later.

___

I believe there is a separate setting for the pin of the PWM buzzer. It's a bit hidden away in the [Device Config](https://meshtastic.org/docs/configuration/radio/device/#gpio-for-pwm-buzzer) section. Have you set this one too?

**Reply by adamstern2000** [reply]
How do you typically configure your devices for external notification with a vibrator and buzzer?

**Reply by todd-herbert** [reply]
With a PWM buzzer, I believe you will need to specify which pin is used in the Device Config section.

![Screenshot_20241029_140346_Meshtastic](https://github.com/user-attachments/assets/7f50daa1-fc5c-4360-8d74-9cc118bc538a)

**Reply by adamstern2000** [reply]
So I tried setting this a few hours ago - it seems to have worked.  I think you are right - on the iphone app it shows up as unset, but the android shows 0.  That must bean that io0 is being activated when the buzzer goes off.  Thanks for pointing this out.

**Comment by adamstern2000** [comment]
I think this is a bug - why would it say shutting down and not shut down?  It shouldn’t say it in the first place, but it if for some reason I soldered something incorrectly, then ut should shut down.  Instead it will say shutting down on the screen until I power it off manually.

**Reply by adamstern2000** [reply]
This should not have been moved from issues.

**Reply by todd-herbert** [reply]
> why would it say shutting down and not shut down?

If the display shows "Shutting down...", this indicated that the user button is held down, and has not been released.
I believe that when the PWM buzzer option is enabled, the pin used is the one set in [Device Config](https://meshtastic.org/docs/configuration/radio/device/#gpio-for-pwm-buzzer), rather than the one set in External notification settings. If you haven't configured this, I imagine the PWM buzzer code is manipulating GPIO0. On the T-Beam S3 Core variant, this is also where the user button is connected, which seems like a likely cause of what you're seeing with the display.

> This should not have been moved from issues.

I believe issues get moved here when they appear to be troubleshooting issues. Your situation does highlight an potential issue though: it is not clear that the the setting in device config must also be set when using a PWM buzzer. It seems like a fairly common source of confusion, and there's probably a good argument for merging the two settings in future.

**Reply by todd-herbert** [reply]
It seems that there is only a [brief note](https://meshtastic.org/docs/configuration/module/external-notification/#gpio-pins-general--led-vibra-and-buzzer) on this in the documentation, other than the [description of the setting](https://meshtastic.org/docs/configuration/radio/device/#gpio-for-pwm-buzzer).

**Reply by adamstern2000** [reply]
Thanks Todd!
